integration_test/third_party_apps_test.go

```diff
@@ -327,6 +328,101 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
+// Returns the authoritative set of all apps available for testing.
+// The authoritative list corresponds to the directory names under
+// integration_test/third_party_apps_data/applications
+func determineAllApps(t *testing.T) map[string]bool {
+	allApps := make(map[string]bool)
+
+	files, err := os.ReadDir("third_party_apps_data/applications")
+	if err != nil {
+		t.Fatalf("got error listing files under third_party_apps_data/applications: %v", err)
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			allApps[file.Name()] = true
+		}
+	}
+	log.Printf("all apps: %v", allApps)
+	return allApps
+}
+
+func modifiedFiles(t *testing.T) []string {
+	cmd := exec.Command("git", "diff", "--name-only", "origin/master")
+	out, err := cmd.Output()
+	log.Printf("git diff output:\n\tstdout:%v\n\tstderr:%v\n", out, err)
+	if err != nil {
+		t.Fatalf("got error calling `git diff`: %v", err)
+	}
+
+	return strings.Split(string(out), "\n")
+}
+
+// Determine what apps are impacted by current code changes.
```

Comment:
For cases when a broken change is introduced in the code that are not in either these two categories, but the code change touches the flow of those third party integration. It will skip tests for such cases..?
I think the code for 3rd party isn’t secluded only to those two folders, it uses many from other places too.

Reply:
By `Postsubmits should omit the -short flag and run the full suite.` @stati0n do you mean the test being kicked before merging into master?

Reply:
Postsubmits means the build for release.  In this change, we are consciously choosing lower coverage in exchange for higher reliability and faster runtime.